src/controller/java/AndroidCallbacks.cpp

```diff
@@ -224,6 +223,32 @@ void ReportCallback::OnReportEnd()
     VerifyOrReturn(!env->ExceptionCheck(), env->ExceptionDescribe());
 }
 
+CHIP_ERROR ConvertReportTlvToJson(const uint32_t & id, TLV::TLVReader * apData, std::string & json)
+{
+    TLV::TLVWriter writer;
+    TLV::TLVReader readerForJavaTLV;
+    readerForJavaTLV.Init(*apData);
+    uint32_t size  = 0;
+    // Add the bytes for attribute tag(1:control + 8:tag + 8:length) and structure(1:struct + 1:close container)
```

Comment:
name the 19 to something like `kExtraSpaceForAttributeTag` or similar.

Reply:
It would be nice to place this in a separate source set and have unit tests. This does not need to live inside android-specific code.

Reply:
done, thanks